x-pack/plugins/security_solution/public/common/components/security_route_page_wrapper/index.tsx

```diff
@@ -6,15 +6,17 @@
  */
 
 import React from 'react';
+import { Redirect } from 'react-router-dom';
 import { TrackApplicationView } from '@kbn/usage-collection-plugin/public';
 import type { SecurityPageName } from '../../../../common';
-import { useLinkAuthorized } from '../../links';
+import { useLinkInfo } from '../../links';
 import { NoPrivilegesPage } from '../no_privileges';
 import { useUpsellingPage } from '../../hooks/use_upselling';
 import { SpyRoute } from '../../utils/route/spy_routes';
 
 interface SecurityRoutePageWrapperProps {
   pageName: SecurityPageName;
+  redirectOnMissing?: boolean;
```

Comment:
When switching the navigation version users may be located on a page that does not exist. In this scenario, we should redirect the user to the home page (get started).

Reply:
Not sure how your logic works but just want to point out that, in case a page does not exist in the current solution tree definition, we look for a different solution where the page _might_ exist. If we find it we redirect to that solution. 
Maybe that does not apply in this context but wanted to share it.